src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -27,4 +27,27 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object object) {
+        if (this == object) {
+            return true;
+        }
+        if (object == null) {
+            return false;
+        }
+        if (getClass() != object.getClass()) {
+            return false;
+        }
+        Rectangle rectangle = (Rectangle) object;
+        return this.length.equals(rectangle.length)
+                && this.width.equals(rectangle.width)
+                && this.color.equals(rectangle.color);
+    }
+
+    @Override
+    public int hashCode() {
+        return 31 * (length.hashCode() + width.hashCode()
```

Comment:
length and width may be null, also. and hashcode should not be 0 or any other constant number, if some of the fields is null